I don't understand the CI error.

It fails for Python 3.3:

https://travis-ci.org/marshmallow-code/marshmallow/jobs/439615371

```
$ pip install -U six
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.3.6/bin/pip", line 7, in <module>
    from pip._internal import main
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/__init__.py", line 40, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/cli/autocompletion.py", line 8, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/cli/main_parser.py", line 8, in <module>
    from pip._internal.cli import cmdoptions
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/cli/cmdoptions.py", line 20, in <module>
    from pip._internal.utils.hashes import STRONG_HASHES
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/utils/hashes.py", line 10, in <module>
    from pip._internal.utils.misc import read_chunks
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_internal/utils/misc.py", line 21, in <module>
    from pip._vendor import pkg_resources
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pip/_vendor/pkg_resources/__init__.py", line 90, in <module>
    raise RuntimeError("Python 3.4 or later is required")
RuntimeError: Python 3.4 or later is required
```

It used to work, with the same six version:

https://travis-ci.org/marshmallow-code/marshmallow/jobs/431313921

```
$ pip install -U six
Collecting six
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Installing collected packages: six
  Found existing installation: six 1.10.0
    Uninstalling six-1.10.0:
      Successfully uninstalled six-1.10.0
Successfully installed six-1.11.0
```

--- a/AUTHORS.rst
+++ b/AUTHORS.rst
@@ -85,3 +85,4 @@ Contributors (chronological)
 - `@sduthil <https://github.com/sduthil>`_
 - Alisson Silveira `@4lissonsilveira <https://github.com/4lissonsilveira>`_
 - Maxim Novikov `@m-novikov <https://github.com/m-novikov>`_
+- Viktor Kerkez `@alefnula<https://github.com/alefnula>`_

--- a/marshmallow/validate.py
+++ b/marshmallow/validate.py
@@ -55,7 +55,7 @@ class URL(Validator):
                 r'^',
                 r'(' if relative else r'',
                 r'(?:[a-z0-9\.\-\+]*)://',  # scheme is validated separately
-                r'(?:[^:@]+?:[^:@]*?@|)',  # basic auth
+                r'(?:[^:@]+?(:[^:@]*?)?@|)',  # basic auth
                 r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+',
                 r'(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|',  # domain...
                 r'localhost|',  # localhost...

--- a/tests/test_validate.py
+++ b/tests/test_validate.py
@@ -23,6 +23,8 @@ from marshmallow import validate, ValidationError
     'http://123.45.67.8/',
     'http://2001:db8::ff00:42:8329',
     'http://www.example.com:8000/foo',
+    'http://user@example.com',
+    'http://user:pass@example.com',
 ])
 def test_url_absolute_valid(valid_url):
     validator = validate.URL(relative=False)

